lib/bsdlib/bsd_os.c

```diff
@@ -49,12 +50,164 @@ static const nrfx_uarte_t uarte_inst = NRFX_UARTE_INSTANCE(1);
 
 void IPC_IRQHandler(void);
 
-int32_t bsd_os_timedwait(uint32_t context, uint32_t timeout)
+#define THREAD_MONITOR_ENTRIES 10
```

Comment:
Should we increase this to the number of sockets in the system?

Reply:
Hmm, I think we could set it to `BSD_MAX_SOCKET_COUNT`, as it gives more or less overview of the maximum number of threads to expect. Anyway, having more threads in the system then this value should not have any tragic consequences, just a little more computation overhead.

Reply:
Ah, so the oldest entry is discarded if all are taken? How would that play out if a thread is removed from the list as the oldest before it's woken again? I'm probably missing something here.

Anyway, I think we can set the array size to `BSD_MAX_IP_SOCKET_COUNT + BSD_MAX_AT_SOCKET_COUNT`.

Reply:
The only drawback of the thread missing on a list is that it will return once from `bsd_os_timedwait` before actually going to sleep (just as if the RPC counter was higher than the last stored). That's the computation overhead I've mentioned. It's not a big deal if the thread is woken up and is missing on the list, as the value is only checked before going to sleep.

A total maximum number of sockets in the system is not a sum of `BSD_MAX_IP_SOCKET_COUNT ` and `BSD_MAX_AT_SOCKET_COUNT` - it's the higher of these. Please have a look at this description, I think `BSD_MAX_SOCKET_COUNT` is fine here:
https://github.com/NordicPlayground/nrfxlib/blob/80a127c5ea388a6a97931c125559ee3bf849e468/bsdlib/include/bsd_limits.h#L38

Reply:
You're absolutely right, my bad. Will read the actual documentation before commenting next time :)

Reply:
The correct number of context for THREAD_MONITOR_ENTRIES is 10. The timedwait is used for other context than the IP/AT sockets. We could add the number of context to bsd_limits.h in the future, to define number of context that needs timewait. There will be additionally 1 more added soon, ending at 11.
@rlubos , @jtguggedal , for details contact me offline.

Reply:
Thanks for clearing that up!

Reply:
In that case, please extend bsd_limits.h in the future, and I'll use this value. For now let's revert to what we had (10).